src/neo/SmartContract/InteropDescriptor.cs

```diff
@@ -21,7 +22,7 @@ internal InteropDescriptor(string name, MethodInfo handler, long fixedPrice, Cal
             this.Name = name;
             this.Hash = BitConverter.ToUInt32(Encoding.ASCII.GetBytes(name).Sha256(), 0);
             this.Handler = handler;
-            this.Parameters = handler.GetParameters().Select(p => new InteropParameterDescriptor(p)).ToArray();
+            this.Parameters = handler.GetParameters().Select(p => new InteropParameterDescriptor(p)).ToList();
```

Comment:
`Array` is faster

Reply:
Arrays are also writable. Since we were making this property public, I thought it best to protect the contents of the collection. 

Besides, I didn't think performance was a concern here as the only place this property is used (`ApplicationEngine.OnSysCall`) also is using reflection invocation. The slight performance difference between looping over array vs IReadOnlyList for small collections (currently, no service method has more than 4 parameters) is miniscule compared to the significant performance overhead of using Reflection.